var/spack/repos/builtin/packages/openblas/package.py

```diff
@@ -52,16 +52,12 @@ class Openblas(Package):
     patch('make.patch')
 
     @property
-    def blas_libs(self):
+    def libs(self):
         shared = True if '+shared' in self.spec else False
         return find_libraries(
             ['libopenblas'], root=self.prefix, shared=shared, recurse=True
         )
 
-    @property
-    def lapack_libs(self):
```

Comment:
your current design appears to be limited in case of a single provider supplying several things, like `mkl` does for `blas`, `lapack` and `scalapack`. In this case, the libs for the last one (`scalapack`) are certainly different from those used in `blas` and `lapack`.  In other words 
`spec['blas'].libs`, `spec['metis'].libs` and `spec['scalapack'].libs` could all be provided by the same package and potentially could all be different. Note that it's different from what you do with a query parameter.